File: Session10/C++/ImplementStackUsingQueue.cpp
https://leetcode.com/problems/implement-stack-using-queues/

class MyStack {
public:
    queue<int> q;
    MyStack() {
        
    }
    
    void push(int x) {
        
        int size = q.size();
        q.push(x);
        while(size--)
        {
            q.push(q.front());
            q.pop();
        }
    }
    
    int pop() {
        
        int result = q.front();
        q.pop();
        return result;
        
    }
    
    int top() {
        
        return q.front();
        
    }
    
    bool empty() {
        
        return q.empty();
        
    }
};

File: Session10/Java/ImplementStackUsingQueue.java
https://leetcode.com/problems/implement-stack-using-queues/

class MyStack {
    
    Queue<Integer> queue = new LinkedList<>();

    public MyStack() {
        
    }
    
    public void push(int x) {
       int size = queue.size();
       queue.add(x);
       while(size>0) {
           queue.add(queue.poll());
           size--;
       }
    }
    
    public int pop() {
        
        return queue.poll();
        
    }
    
    public int top() {
        
        return queue.peek();
        
    }
    
    public boolean empty() {
        
        return queue.isEmpty();
        
    }
}